dfu: mcuboot: rename boot_swap_type> mcuboot_swap_type

nashif opened the pull request:
This function conflicts with a function of the same name in mcuboot. This happens when building USB DFU support into mcuboot.

DFU over USB uses image manager and mcuboot internals to manage images downloaded to the device.

nvlsianpu commented:
> They conflict if we build the bootloader
with USB DFU.

What a bootloader? It is something else than mcuboot?

nvlsianpu commented:
Honestly - this change is wired - better to change function names in application your attempt to build, or if this is relay impossibly - change names here.

nashif commented:
> What a bootloader? It is something else than mcuboot?

yes, mcuboot.

nashif commented:
> Honestly - this change is wired - better to change function names in application your attempt to build, or if this is relay impossibly - change names here.

dfu usb depends on image manager and pulls mcuboot.c, mcuboot.c duplicates function names in mcuboot, but those are only used when you have the application use mcumgr (which does not make sense to enable in the bootloader), so just use those duplicated functions with mcumgr and not when building mcuboot. I do not see anything weird about that.

nvlsianpu commented:
Don't see such file name in mcuboot mainline - so this is something you working on.

USB DFU (DFU class instead of serial recovery?) in bootloader should use mcuboot primitives instead of the zephyr primitives.

nashif commented:
> Don't see such file name in mcuboot mainline - so this is something you working on.
> 

boot_swap_type is both in Zephyr and in mcuboot.


> USB DFU (DFU class instead of serial recovery?) in bootloader should use mcuboot primitives instead of the zephyr primitives.

it is up to me to decide how I want to build my bootloader and what features I want to add in.

mbolivar commented on subsys/dfu/boot/mcuboot.c:
```diff
@@ -349,6 +353,7 @@ int boot_read_bank_header(u8_t area_id,
 	return 0;
 }
 
+#ifdef CONFIG_MCUMGR
```
I don't understand. These are static and there are no functions by the same name anywhere else in the tree. Why is this necessary?

mbolivar replied:
Missed the thread on this.

Can we just rename them in this file to not conflict or something, since they're static? No need for extra ifdefing.

nashif replied:
yes, renaming is probably a better option, I thought there might be some purpose behind the same names. Btw, we have functions defined in the is file that are used no where in the tree. dead code basically, i.e `boot_read_bank_header`

nashif replied:
> since they're static?

btw, the main culprit is not static but if we ifdef it, the others will be become 'unused', hence why they all were ifdef'ed

nvlsianpu replied:
> yes, renaming is probably a better option. I thought there might be some purpose behind the same names.

That is my opinion as well, so will approve. Name were the same as function does the same as in mucboot (and had no better name before) - only that.

nashif committed:
dfu: mcuboot: rename boot_swap_type> mcuboot_swap_type

This function conflicts with a function of the same name in mcuboot.
This happens when building USB DFU support into mcuboot.

DFU over USB uses image manager and mcuboot internals to manage images
downloaded to the device.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

mbolivar commented on samples/subsys/usb/dfu/README.rst:
```diff
@@ -86,7 +86,7 @@ showing this output to the console:
   [MCUBOOT] [INF] boot_status_source: Scratch: magic=unset, copy_done=0xe, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Boot source: slot 0
-  [MCUBOOT] [INF] boot_swap_type: Swap type: test
+  [MCUBOOT] [INF] mcuboot_swap_type: Swap type: test
```
This should not have been changed; those log messages are from mcuboot itself

nvlsianpu replied:
fix covered in #13552

mbolivar commented on samples/subsys/usb/dfu/README.rst:
```diff
@@ -102,7 +102,7 @@ USB DFU sample, showing this output to the console:
   [MCUBOOT] [INF] boot_status_source: Scratch: magic=unset, copy_done=0xe, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Boot source: none
-  [MCUBOOT] [INF] boot_swap_type: Swap type: revert
+  [MCUBOOT] [INF] mcuboot_swap_type: Swap type: revert
```
same here

nvlsianpu replied:
fix covered in #13552
